ert3/storage/_storage.py

```diff
@@ -96,45 +111,147 @@ def get_experiment_names(*, workspace: Path) -> Set[str]:
     return experiment_names
 
 
-def _add_data(
-    workspace: Path, experiment_name: str, record_name: str, data: Any
-) -> None:
+def _get_record_type(ensemble_record: ert3.data.EnsembleRecord) -> ert3.data.RecordType:
+    record_type = ensemble_record.records[0].record_type
+    for record in ensemble_record.records:
```

Comment:
Doesnt this type of validation belong in the EnsembleRecord class?

Reply:
It does... But as mentioned above there is currently no requirement for uniform record type over an ensemble record. I suggest we make an issue on it (as listed under future work) and act on it promptly.